STAR-Delegation/CSR-template/template-schema.cddl

```diff
@@ -0,0 +1,81 @@
+csr-template-schema = {
+  keyTypes: [ 1* $keyType ]
+  ? subject: distinguishedName
+  extensions: extensions
+}
+
+mandatory-wildcard = "**"
+optional-wildcard = "*"
+wildcard = mandatory-wildcard / optional-wildcard
+
+; regtext matches all text strings but "*" and "**"
+regtext = text .regexp "([^\*].*)|([\*][^\*].*)|([\*][\*].+)"
+
+regtext-or-wildcard = regtext / wildcard
+
+distinguishedName = {
+  ? country: regtext-or-wildcard
+  ? stateOrProvince: regtext-or-wildcard
+  ? locality: regtext-or-wildcard
+  ? organization: regtext-or-wildcard
+  ? organizationalUnit: regtext-or-wildcard
+  ? emailAddress: regtext-or-wildcard
+  ? commonName: regtext-or-wildcard
+}
+
+$keyType /= rsaKeyType
+$keyType /= ecKeyType
+
+rsaKeyType = {
+  PublicKeyType: "RSA"
+  PublicKeyLength: $rsaKeySize
+  SignatureType: $signatureType / mandatory-wildcard
+}
+
+$rsaKeySize /= 2048
+$rsaKeySize /= 4096
+$rsaKeySize /= 8192
+
+$signatureType /= "sha256WithRSAEncryption"
+
+ecKeyType = {
+  PublicKeyType: "ecPublicKey"
+  ? Curve: $ecCurve ; not populated for Ed25519
+  SignatureType: $ecSignatureType
+}
+
+$ecCurve /= "secp521r1"
```

Comment:
@yaronf Regarging the choice of secp521r1.  What was the rationale? (It doesn't seem to me to be the most popular/interoperable.)

Reply:
I just picked an example and may definitely have got it wrong. I just want to cover whatever ecdsa certs are out there in the real world.

Reply:
> I just picked an example and may definitely have got it wrong. I just want to cover whatever ecdsa certs are out there in the real world.

OK, I think we should probably have at least Section 6.1.5 of https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.7.3.pdf mapped

Reply:
Makes sense. So Sec. 7.1.3.1.2 of the same document.

Reply:
done in d304669